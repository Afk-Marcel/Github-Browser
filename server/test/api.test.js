const request = require("supertest");
const app = require("../index");

describe("GET /api/users/:username", () => {
  let server;
  const port = 5002;

  beforeAll((done) => {
    // Start the server and signal Jest that setup is complete
    server = app.listen(port, () => {
      console.log(`Test server running on port ${port}`);
      done();
    });
  });

  afterAll((done) => {
    // Close the server and signal Jest that cleanup is complete
    server.close(() => {
      console.log(`Test server on port ${port} closed`);
      done();
    });
  });

  it("should match the snapshot", async () => {
    const response = await request(server).get("/api/users/Afk-Marcel");
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});
