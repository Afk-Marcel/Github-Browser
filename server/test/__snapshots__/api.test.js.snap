// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /api/users/:username should match the snapshot 1`] = `
{
  "avatar_url": "https://avatars.githubusercontent.com/u/156346612?v=4",
  "bio": "18 year old who loves what he does!",
  "blog": "",
  "collaborators": 0,
  "company": null,
  "created_at": "2024-01-11T16:27:49Z",
  "disk_usage": 10557,
  "email": null,
  "events_url": "https://api.github.com/users/Afk-Marcel/events{/privacy}",
  "followers": 1,
  "followers_url": "https://api.github.com/users/Afk-Marcel/followers",
  "following": 1,
  "following_url": "https://api.github.com/users/Afk-Marcel/following{/other_user}",
  "gists_url": "https://api.github.com/users/Afk-Marcel/gists{/gist_id}",
  "gravatar_id": "",
  "hireable": null,
  "html_url": "https://github.com/Afk-Marcel",
  "id": 156346612,
  "location": null,
  "login": "Afk-Marcel",
  "name": "Marcel Greyling",
  "node_id": "U_kgDOCVGo9A",
  "organizations_url": "https://api.github.com/users/Afk-Marcel/orgs",
  "owned_private_repos": 2,
  "plan": {
    "collaborators": 0,
    "name": "free",
    "private_repos": 10000,
    "space": 976562499,
  },
  "private_gists": 0,
  "public_gists": 0,
  "public_repos": 4,
  "received_events_url": "https://api.github.com/users/Afk-Marcel/received_events",
  "repos_url": "https://api.github.com/users/Afk-Marcel/repos",
  "site_admin": false,
  "starred_url": "https://api.github.com/users/Afk-Marcel/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/Afk-Marcel/subscriptions",
  "total_private_repos": 2,
  "twitter_username": null,
  "two_factor_authentication": false,
  "type": "User",
  "updated_at": "2024-08-02T17:03:25Z",
  "url": "https://api.github.com/users/Afk-Marcel",
}
`;
